### PR TITLE
USSD public line update message receiver flow

### PR DESCRIPTION
**Pull request by NicolaasDieBaas**
two new states to be added to the public line that calls for change to the flow

### CODE REVIEW

**Comment by NicolaasDieBaas** [comment]
@gsvr please review

File: test/ussd_public.test.js [hunk 1]
```diff
@@ -151,7 +160,7 @@ describe("Hello Mama app", function() {
                         .run();
                 });
                 // assuming flow via unregistered user...
-                it("to state_msisdn_not_recognised", function() {  //st-F
+                it.skip("to state_msisdn_not_recognised", function() {  //st-F
```
Comment (gsvr) [inline test/ussd_public.test.js:163]: Did you forget the `.skip` here?

File: test/ussd_public.test.js [hunk 2]
```diff
@@ -482,6 +510,26 @@ describe("Hello Mama app", function() {
...
+                            , '1'  // state_msisdn_permission - yes
+                            , '5'  // state_main_menu - stop receiving messages
+                            , '5'  // state_optout_reason - other
+                        )
+                        .check.interaction({
+                            state: 'state_optout_receiver',
+                            reply: [
+                                "Who would you like to stop receiving messages?",
+                                "1. Only me",
+                                "2. The Father",
```
Comment (gsvr) [inline test/ussd_public.test.js]: We need to add another state here that does not give option 2 when the role player is the father.

**Comment by gsvr** [comment]
@NicolaasDieBaas Done for now.

File: test/ussd_public.test.js (hunk 1 above)
Comment (NicolaasDieBaas) [inline test/ussd_public.test.js:163]: We agreed to intentionally do this.

**Comment by NicolaasDieBaas** [comment]
@gsvr Second-pass please.  There might be a more elegant way of displaying state 16

**Comment by gsvr** [comment]
:+1:

**Pull request merged**